lib/model/folder_sendrecv.go

```diff
@@ -1548,6 +1587,34 @@ func (f *sendReceiveFolder) performFinish(file, curFile protocol.FileInfo, hasCu
 	return nil
 }
 
+// finalizeEncrypted adds a trailer to the encrypted file containing the
+// serialized FileInfo and the length of that FileInfo. When initializing a
+// folder from encrypted data we can extract this FileInfo from the end of
+// the file and regain the original metadata.
+func (f *sendReceiveFolder) finalizeEncrypted(file protocol.FileInfo, tempName string) (protocol.FileInfo, error) {
+	size := file.ProtoSize()
+	bs := make([]byte, 4+size)
+	n, err := file.MarshalTo(bs)
+	if err != nil {
+		return protocol.FileInfo{}, err
+	}
+	binary.BigEndian.PutUint32(bs[n:], uint32(n))
+	bs = bs[:n+4]
+
+	fd, err := f.fs.OpenFile(tempName, fs.OptWriteOnly|fs.OptAppend, 0600)
```

Comment:
Should we defer closing the file in shared puller state or something, so we wouldn't need to reopen it?